src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/CollectLogsActionTests.cs

```diff
@@ -0,0 +1,347 @@
+                    LogsTestUtilities.ValidateEntry(LogsTestUtilities.Category3WarningEntry, await reader.ReadAsync());
+                    LogsTestUtilities.ValidateEntry(LogsTestUtilities.Category3ErrorEntry, await reader.ReadAsync());
+                    LogsTestUtilities.ValidateEntry(LogsTestUtilities.Category3CriticalEntry, await reader.ReadAsync());
+                    Assert.False(await reader.WaitToReadAsync());
+                },
+                logFormat);
+        }
+
+        private async Task ValidateLogsActionAsync(
+            LogsConfiguration configuration,
+            Func<ChannelReader<LogEntry>, Task> callback,
+            LogFormat logFormat)
+        {
+            using TemporaryDirectory tempDirectory = new(_outputHelper);
+
+            await TestHostHelper.CreateCollectionRulesHost(_outputHelper, rootOptions =>
+            {
+                rootOptions.AddFileSystemEgress(ExpectedEgressProvider, tempDirectory.FullName);
+
+                rootOptions.CreateCollectionRule(DefaultRuleName)
+                    .AddCollectLogsAction(ExpectedEgressProvider, out CollectLogsOptions collectLogsOptions)
+                    .SetStartupTrigger();
+
+                collectLogsOptions.Duration = CommonTestTimeouts.LogsDuration;
+                collectLogsOptions.FilterSpecs = configuration.FilterSpecs;
+                collectLogsOptions.DefaultLevel = configuration.LogLevel;
+                collectLogsOptions.Format = logFormat;
+                collectLogsOptions.UseAppFilters = configuration.UseAppFilters;
+            }, async host =>
+            {
+                IOptionsMonitor<CollectionRuleOptions> ruleOptionsMonitor = host.Services.GetService<IOptionsMonitor<CollectionRuleOptions>>();
+                CollectLogsOptions options = (CollectLogsOptions)ruleOptionsMonitor.Get(DefaultRuleName).Actions[0].Settings;
+
+                // This is reassigned here due to a quirk in which a null value in the dictionary has its key removed, thus causing LogsDefaultLevelFallbackActionTest to fail. By reassigning here, any keys with null values are maintained.
+                options.FilterSpecs = configuration.FilterSpecs;
+
+                ICollectionRuleActionFactoryProxy factory;
+                Assert.True(host.Services.GetService<ICollectionRuleActionOperations>().TryCreateFactory(KnownCollectionRuleActions.CollectLogs, out factory));
+
+                using CancellationTokenSource endpointTokenSource = new CancellationTokenSource(CommonTestTimeouts.LogsTimeout);
+
+                EndpointInfoSourceCallback endpointInfoCallback = new(_outputHelper);
+                await using var source = _endpointUtilities.CreateServerSource(out string transportName, endpointInfoCallback);
+                source.Start();
+
+                AppRunner runner = _endpointUtilities.CreateAppRunner(transportName, TargetFrameworkMoniker.Net60); // Arbitrarily chose Net60; should we test against other frameworks?
```

Comment:
Make that TargetFrameworkMoniker a parameter of the test so that tests can be parameterized to (output format) x (target framework moniker). You might want to use MemberData attribute on the tests rather than the InlineData so that you compute the list in one place and allows us to more easily add/remove formats and TFMs.

Reply:
I'm working on it right now; it doesn't look like it'll take too long. Not for this PR, but for a QFW one, should I also be doing something similar for the other actions (they're currently all just using net6.0 by default)?

Reply:
Yup, that's why I said you can defer this one, because the others are fixed only to net6.0